titus-ext/aws/src/main/java/com/netflix/titus/ext/aws/loadbalancer/AwsLoadBalancerConnector.java

```diff
@@ -173,14 +175,19 @@ public Completable isValid(String loadBalancerId) {
                     Set<String> ips = result.getTargetHealthDescriptions().stream()
                             .map(description -> description.getTarget().getId())
                             .collect(Collectors.toSet());
-                    return new LoadBalancer(id, LoadBalancer.State.ACTIVE, ips);
+                    return new LoadBalancer(loadBalancerId, LoadBalancer.State.ACTIVE, ips);
                 })
                 .onErrorResumeNext(throwable -> {
                     connectorMetrics.failure(AwsLoadBalancerConnectorMetrics.AwsLoadBalancerMethods.DescribeTargetHealth, throwable, startTime);
                     if (throwable instanceof TargetGroupNotFoundException) {
-                        return Single.just(new LoadBalancer(id, LoadBalancer.State.REMOVED, Collections.emptySet()));
+                        return Single.just(new LoadBalancer(loadBalancerId, LoadBalancer.State.REMOVED, Collections.emptySet()));
                     }
                     return Single.error(throwable);
                 });
     }
+
+    private AmazonElasticLoadBalancingAsync getClient(String loadBalancerId) {
+        Arn arn = Arn.fromString(loadBalancerId);
+        return clientProvider.getLoadBalancingClient(arn.getAccountId());
```

Comment:
Is it an invariant that account id is always set in the load balancer id?

Reply:
yes, a valid ARN must always contain the accountId and `Arn.fromString(loadBalancerId)` does the validation.